corehq/apps/sso/models.py

```diff
@@ -144,6 +144,28 @@ def domain_has_editable_identity_provider(cls, domain):
         owner = BillingAccount.get_account_by_domain(domain)
         return cls.objects.filter(owner=owner, is_editable=True).exists()
 
+    @classmethod
+    def get_active_identity_provider_by_username(cls, username):
+        """
+        Returns the active Identity Provider associated with a user's email
+        domain or None.
+        :param username: (string)
+        :return: IdentityProvider or None
+        """
+        try:
+            email_domain = username.split('@')[1]
```

Comment:
i do actually want to throw an `IndexError` if it's poorly formatted, because then it's not a real username...`[-1]` would never throw an `IndexError` which is not what I want either.